pkg/telemetry/marketing/device.go

```diff
@@ -0,0 +1,63 @@
+package marketing
+
+import (
+	"context"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/env"
+	"github.com/stackrox/rox/pkg/version"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+const annotation = "rhacs.redhat.com/telemetry-apipaths"
+const orgID = "rhacs.redhat.com/organization-id"
+
+var config *Config
+
+// GetDeviceConfig collects the central instance telemetry configuration.
+func GetDeviceConfig() (*Config, error) {
+	if config != nil {
+		return config, nil
+	}
+	rc, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, errors.Wrap(err, "cannot create k8s config")
+	}
+	clientset, err := kubernetes.NewForConfig(rc)
+	if err != nil {
+		return nil, errors.Wrap(err, "cannot create k8s clientset")
+	}
+	v, err := clientset.ServerVersion()
+	if err != nil {
+		return nil, err
+	}
+	orchestrator := storage.ClusterType_KUBERNETES_CLUSTER.String()
+	if env.OpenshiftAPI.BooleanSetting() {
+		orchestrator = storage.ClusterType_OPENSHIFT_CLUSTER.String()
+	}
+
+	di := clientset.AppsV1().Deployments(env.Namespace.Setting())
+	opts := v1.GetOptions{}
+	d, err := di.Get(context.Background(), "central", opts)
+	if err != nil {
+		return nil, errors.Wrap(err, "cannot get central deployment")
+	}
+	paths := d.GetAnnotations()[annotation]
+
+	config = &Config{
```

Comment:
We discussed using the same hashing algorithm on the backend and frontend to combine the central ID and Org ID to create an obfuscated Segment identifier.

I propose that the hash happens once in the API, and is passed to the FE along with this info. Doing the same thing in two different place invites problems to arise. 
https://en.wikipedia.org/wiki/Murphy%27s_law

Stephan's question below about what to do when Org ID is not available, begs a follow-up question. If API passes the hash, does central ID + "" still create a valid, unique hash to use in Segment?

Reply:
I'd avoid hashing non-user IDs, as they could be of a use in clear. I'd also avoid combining IDs to a single hash, as we might want to correlate different IDs separately.
User IDs could be hashed by central, indeed.

Reply:
Are we sure that user IDs can never be the same for two different instances of Central ?

As @rukletsov mentioned in another comment, we could send user IDs and org IDs, along with other optional useful metadata, in an attached dictionary.

Reply:
We shall not assume uniqueness of user IDs across Centrals but that's fine as long as we use `<CentralID, UserID>` to identify universally unique ACS users.